### PR TITLE
Clean up internal/testutil/helpers package.

### PR DESCRIPTION
**Pull request by matthewdale**
Improvements:
* Update package name at `internal/testutil/helpers` to align with import path (i.e. rename package to `helpers`).
* Move all single-use functions in `helpers` package to where they are used and un-export them.
* Remove duplicate test assertion functions from `helpers` package.
* Add `t.Helper()` to the one remaining test helper function.

### CODE REVIEW

File: internal/testutil/helpers/helpers.go [hunk 1]
```diff
@@ -4,63 +4,21 @@
...
-
-// RequireNotNil throws an error if var is nil
-func RequireNotNil(t *testing.T, variable interface{}, msgFormat string, msgVars ...interface{}) {
-	if IsNil(variable) {
-		t.Fatalf(msgFormat, msgVars...)
-	}
-}
-
-// RequireNil throws an error if var is not nil
-func RequireNil(t *testing.T, variable interface{}, msgFormat string, msgVars ...interface{}) {
```
Comment (benjirewis) [inline internal/testutil/helpers/helpers.go:55]: Some strange helpers here; makes much more sense to just use the `require` package...

**Review by benjirewis — APPROVED** [review]
This is a great cleanup; thanks!

File: internal/testutil/helpers/helpers.go [hunk 2]
```diff
@@ -4,63 +4,21 @@
 // not use this file except in compliance with the License. You may obtain
 // a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 
-package testhelpers // import "go.mongodb.org/mongo-driver/internal/testutil/helpers"
```
[the remark this answers is not in the document]
Comment (matthewdale) [reply internal/testutil/helpers/helpers.go:7]: I assume it's vestigial (i.e. maybe the directory used to be `testhelpers`), but I can't find anything in the git history where it was different.

File: internal/testutil/helpers/helpers.go (hunk 1 above)
Comment (matthewdale) [reply internal/testutil/helpers/helpers.go:55]: Agreed. I'm assuming again that these are vestigial and were never removed when we started using the [stretchr/testify](https://github.com/stretchr/testify) package and the internal `assert` package.

File: mongo/integration/unified/collection_data.go [hunk 3]
```diff
@@ -71,13 +70,22 @@ func (c *collectionData) createCollection(ctx context.Context) error {
...
-	docs := testhelpers.RawSliceToInterfaceSlice(c.Documents)
+	docs := rawSliceToInterfaceSlice(c.Documents)
 	if _, err := coll.InsertMany(ctx, docs); err != nil {
 		return fmt.Errorf("error inserting data: %v", err)
 	}
 	return nil
 }
 
+// rawSliceToInterfaceSlice converts a []bson.Raw to []interface{}.
+func rawSliceToInterfaceSlice(elems []bson.Raw) []interface{} {
```
[the remark this answers is not in the document]
Comment (matthewdale) [reply mongo/integration/unified/collection_data.go:81]: The current behavior of `helpers.RawToInterfaceSlice` extracts the `[]bson.RawValue` from the input document, converts each of those raw values to a `bson.Raw` document, and adds them to a `[]interface{}`. The behavior of `rawSliceToInterfaceSlice` just adds the `bson.Raw` documents to a `[]interface{}`. If we used the proposed function, all current callers of `helpers.RawToInterfaceSlice` would have to do that `bson.Raw` document to `[]bson.Raw` documents conversion themselves, which could add a lot of code.

That could work with two functions:
```go
func RawToDocuments(doc bson.Raw) []bson.Raw
func RawToInterfaceSlice(docs ...bson.Raw) []interface{}
```
Converting a `bson.Raw` document that contains a sequence of documents could be done like
```go
RawToInterfaceSlice(RawToDocuments(doc)...)
```

I think that change makes sense. I'll add those two functions and update the existing `RawToInterfaceSlice` calls to use them.

[the remark this answers is not in the document]
Comment (prestonvasquez) [reply mongo/integration/unified/collection_data.go]: Nice! Thanks for the explanation.

**Review by prestonvasquez — APPROVED** [review]

**Pull request merged**